File: shortestworddistance.java
//Java Solution

class Solution {
    public int shortestDistance(String[] words, String word1, String word2) {
        int i1 = -1;
        int i2 = -1;
        int ans = Integer.MAX_VALUE;
        
        for (int i = 0; i < words.length; i++) {
            if (words[i].equals(word1)) {
                i1 = i;
            } else if (words[i].equals(word2)) {
                i2 = i;
            }
            
            if (i1 != -1 && i2 != -1) {
                ans = Math.min(ans, Math.abs(i2 - i1));
            }
        }
        
        return ans;
    }
}

File: shortestworddistance.js
//Objective is to find the shortest distance in a 1-D array between 2 words.

let arr = ["practice", "makes", "perfect", "coding", "makes"]
let word1 = "practice"
let word2 = "makes"


//O(n) solution that updates the minimum distance during each iteration.

let index1 = -1
let index2 = -2
let result = Infinity

for (let i = 0; i < arr.length; i++) {
    if (arr[i] == word1) {
        index1 = i
    } else if (arr[i] == word2) {
        index2 = i
    }

    if (index1 !== -1 && index2 !== -1) {
        result = Math.min(result, Math.abs(index2 - index1))
    }
}

return result